src/layouts/components/Editor/extensions/Image/Image.ts

```diff
@@ -55,22 +55,16 @@ export const IsomerImage = Image.extend({
           return false
         }
 
-        if (dispatch) {
-          const node = editor.schema.nodes.image.create(options)
-
-          if (tr.selection.empty) {
-            const offset = tr.selection.to
-
-            tr.insert(offset, node)
-              .scrollIntoView()
-              .setSelection(Selection.near(tr.doc.resolve(offset)))
-          } else {
-            const offset = tr.selection.to + 1
+        const { from, to } = tr.selection
 
-            tr.insert(offset, node)
-              .scrollIntoView()
-              .setSelection(Selection.near(tr.doc.resolve(offset)))
+        tr.doc.nodesBetween(from, to, (node, pos) => {
+          if (node.type === editor.schema.nodes.image) {
+            tr.setNodeMarkup(pos, null, { ...node.attrs, ...options })
```

Comment:
fix here is fine, but it might remove image _creation_ functionality as we assume that the node between `from, to` is already an `image` (we do an `if` check) 

we should rebase this off your branch and rename this `createImage` so intended usage is clear + shift all usages _outside_ of `EditPage` (alt text, hyper link, resize) to new commands.

i think the new command for `alt text/hyperlink` can be called `setImageMeta` (or just create two small commands that do the same to avoid devs lumping everything into those commands).

this solves the problem + sets a good precedent for code organisation to avoid future bugs popping out.

Reply:
Oh yeah! Thanks @seaerchin you are right. It affects the image creation, I should have tested it more thoroughly, my bad. 
I will fix it and also separate the functionality of editing image from creating image in the commands. 

Will update once the new commit is pushed.

Reply:
Sure, refactored in 5be2b182bc3a214eeedc184e83bb73ea29bf32be

I added a method called `setImageMeta` to handle all the top level attributes change of image node, including `hylerLink` and `altText`. Decided to not create small commands for those top-level attributes as no clear benefits. The separation of `style` tag is useful because we can enforce the validity of `width` value inside the `style` string. For top-level attributes, we already have types defined in `IsomerImageOptions`.